sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -159,40 +159,38 @@ object GpuHashJoin extends Arm {
    * Filter rows from the batch where any of the keys are null.
    */
   def filterNulls(cb: ColumnarBatch, boundKeys: Seq[Expression]): ColumnarBatch = {
-    var mask: ai.rapids.cudf.ColumnVector = null
+    var mask: Option[ColumnVector] = None
     try {
       withResource(GpuProjectExec.project(cb, boundKeys)) { keys =>
         val keyColumns = GpuColumnVector.extractBases(keys)
-        keyColumns.foreach { column =>
-          if (column.hasNulls) {
-            withResource(column.isNotNull) { nn =>
-              if (mask == null) {
-                mask = nn.incRefCount()
-              } else {
-                mask = withResource(mask) { _ =>
-                  mask.and(nn)
-                }
+        mask = keyColumns
+          .iterator
+          .filter(_.hasNulls())
+          .map(_.isNotNull())
+          .foldLeft(None: Option[ColumnVector])((acc, nn) => {
```

Comment:
I would prefer to keep it unchanged here. `foldLeft` holds an internal intermidate result (`acc`) which should be always closed. But this will not happen if any exception happens during the fold processing, because it is not exposed by specifying it to the `mask` variable.